examples/with-federated-authentication/src/pages/index.tsx

```diff
@@ -0,0 +1,162 @@
+import Image from "next/image";
+import styles from "./index.module.css";
+import { getWebAuthnAttestation, TurnkeyApi } from "@turnkey/http";
+import { useForm } from "react-hook-form";
+import axios from "axios";
+import { buffer } from "stream/consumers";
+import { useState } from "react";
+
+type subOrgFormData = {
+  orgName: string;
+};
+
+type privateKeyFormData = {
+  privateKeyName: string;
+};
+
+const es256 = -7;
+const publicKey = "public-key";
+
+const generateRandomBuffer = (): ArrayBuffer => {
+  const arr = new Uint8Array(32);
+  crypto.getRandomValues(arr);
+  return arr.buffer;
+};
+
+export default function Home() {
+  const [subOrgId, setSubOrgId] = useState<string | null>(null);
+  const { register: subOrgFormRegister, handleSubmit: subOrgFormSubmit } =
+    useForm<subOrgFormData>();
+  const {
+    register: privateKeyFormRegister,
+    handleSubmit: privateKeyFormSubmit,
+  } = useForm<privateKeyFormData>();
+
+  const createPrivateKey = async (data: privateKeyFormData) => {
+    if (!subOrgId) {
+      throw new Error("sub org id id not found");
+    }
+
+    const federatedRequest = await TurnkeyApi.federatedPostCreatePrivateKeys({
+      body: {
+        type: "ACTIVITY_TYPE_CREATE_PRIVATE_KEYS",
+        organizationId: subOrgId,
+        timestampMs: String(Date.now()),
+        parameters: {
+          privateKeys: [
+            {
+              privateKeyName: data.privateKeyName,
+              curve: "CURVE_SECP256K1",
+              addressFormats: ["ADDRESS_FORMAT_ETHEREUM"],
+              privateKeyTags: [],
+            },
+          ],
+        },
+      },
+    });
+
+    await axios.post("/api/proxyRequest", federatedRequest);
+
+    alert(`Hooray! Key ${data.privateKeyName} created.`);
+  };
+
+  const createSubOrg = async (data: subOrgFormData) => {
+    const challenge = generateRandomBuffer();
+    const authenticatorUserId = generateRandomBuffer();
+
+    const attestation = await getWebAuthnAttestation({
+      publicKey: {
+        rp: {
+          id: "localhost",
+          name: "Turnkey WebaAuthn Demo",
+        },
+        challenge,
+        pubKeyCredParams: [
+          {
+            type: publicKey,
+            alg: es256,
+          },
+        ],
+        user: {
+          id: authenticatorUserId,
+          name: data.orgName,
+          displayName: data.orgName,
+        },
+      },
+    });
+
+    const res = await axios.post("/api/subOrg", {
+      orgName: data.orgName,
+      attestation,
+      // The challenge must be base64 url encoded
+      challenge: Buffer.from(challenge)
+        .toString("base64")
+        .replace(/\+/g, "-")
+        .replace(/\//g, "_")
+        .replace(/=/g, ""),
+    });
```

Comment:
let's extract this into a function for readability: `challenge: base64urlencode(challenge)`, and then you can remove the comment because what's going on is pretty clear.